platform/darwin/src/MGLFeature.mm

```diff
@@ -177,6 +177,18 @@ - (NSDictionary *)geoJSONDictionary {
     return mbgl::Feature{geometry};
 }
 
+- (mbgl::FeatureCollection)mbglFeatureCollection {
+    mbgl::FeatureCollection featureCollection;
+    featureCollection.reserve(self.shapes.count);
+    for (id <MGLFeaturePrivate> feature in self.shapes) {
+        if ([feature isMemberOfClass:[MGLShapeCollectionFeature class]]) {
+            featureCollection.push_back([self mbglFeatureCollection]);
+        }
+        featureCollection.push_back([feature mbglFeature]);
```

Comment:
MGLShapeCollectionFeature raises an exception upon `-mbglFeature`. I guess this should be in an else statement?